dashboard/app/controllers/application_controller.rb

```diff
@@ -334,8 +334,8 @@ def prevent_caching
     # Check that the child account policy is currently enabled.
     return unless ::Cpa.cpa_experience(request)
 
-    # Anonymous users are NOT affected by our Child Account Policy
-    return unless current_user
+    # Lock the user out if they do not comply with CAP during the lockout phase.
+    Services::ChildAccount.lock_out(current_user) if Policies::ChildAccount.lockable?(current_user)
```

Comment:
What do you think about creating a method like `Services::ChildAccount.refresh` where all our transition logic is contained? Otherwise most of the conditional transition logic will end up living here, in the ApplicationController, for example if the user should be transitions from `nil` -> `grace_period`. We could even add a transition from `[grace_period, locked_out]` -> `permission_granted`. Then all our unit tests can be in one place and the rest of the code base can simply call `refresh` and assume the user in in the correct CPA state.

By the way, I'm not asking for you to add all the other transition to `refresh`, yet. I'm only asking you to consider starting the pattern so we can follow it with the other transitions.

Reply:
By the way, feel free to choose a better name, I was having trouble thinking of a good one.

Reply:
Correct, I plan to implement this in the upcoming tasks related to the "grace period"

Reply:
Correct, I'm going to implement this in the upcoming tasks related to the "grace period"